### ISSUE TITLE
Note on high memory consumption

### ISSUE THREAD

**Issue by davoodwadi** [InstantID/InstantID#135]
I noticed most memory saving tricks don't help it run on 5GB VRAM. So here are the tricks to help:

1. Most importantly, note that by default the pipeline outputs 2000 x 2000 px. So set the height and width to the original SDXL:
```
image = pipe(
    prompt,
    image_embeds=face_emb,
    image=face_kps,
    controlnet_conditioning_scale=0.8,
    ip_adapter_scale=0.8,
    num_inference_steps=num_inference_steps,
    guidance_scale=guidance_scale,
    height=1024,
    width=1024
).images[0]
```

2. Putting the FaceAnalysis object in gpu is an overkill. To save memory put in on CPU:
```
app = FaceAnalysis(name='antelopev2', root='./', providers=['CPUExecutionProvider']) 
```
3. Try to do the image prompt encoding on the CPU and pass the result to the GPU later. This part is involved in terms of code.

**Comment by FurkanGozukara** [comment]
they need to fix cpu off loading i reported many times by devs ignoring

https://github.com/InstantID/InstantID/issues/34

**Comment by camoody1** [comment]
> they need to fix cpu off loading i reported many times by devs ignoring
> 
> #34

I read your posts on the other issue. Is there some code change I can make to one of the files to make the ComfyUI version of InstantID run better on my 3060 12GB card?

**Comment by FurkanGozukara** [comment]
> > they need to fix cpu off loading i reported many times by devs ignoring
> > #34
> 
> I read your posts on the other issue. Is there some code change I can make to one of the files to make the ComfyUI version of InstantID run better on my 3060 12GB card?

i dont know comfyui but i updated my patreon standalone graido app with cpu offloading

now working great on 12 GB gpu

https://www.patreon.com/posts/1-click-very-gui-97769887

**Comment by al-swaiti** [comment]
> ?

Yes you can ,,,, firstly try full image of the face no more details,,,, I edited the code I will upload it next,hour I'm outside

**Comment by rupeshs** [comment]
Here is CPU version, only 12GB system RAM is required https://github.com/rupeshs/instantidcpu
